### DISCUSSION TITLE
Cursos.rownumber strange behaviour

### DISCUSSION THREAD

**Discussion by pavel-tarasov** [psycopg/psycopg#436] [Q&A]
I experienced the behaviour of the attribute rownumber of Cursor class. In [doc](https://www.psycopg.org/psycopg3/docs/api/cursors.html#psycopg.Cursor.rownumber) I see that it should be `None` if there is no result to fetch. But in my example, I see 0 in this attribute after running the query, which doesn't have anything to fetch ("insert" without returning). And same behaviour when there is what to fetch from the query ("insert" with returning). So, it looks like I can't use this attribute to check if I can catch something from the query.
Question: is it correct behaviour or not? If yes, should we change the documentation?

**Comment by pavel-tarasov** [comment]
Attached code example.
[rownumber_example.txt](https://github.com/psycopg/psycopg/files/9971660/rownumber_example.txt)

**Comment by dvarrazzo** [comment]
Yes, I think it makes sense to fix it.

At the moment it return unconditionally a number when there is a result, regardless of whether the result contains tuples or not.

**Reply by pavel-tarasov** [reply]
Ok, opened an issue.
